src/tableau_server/tableau_server_objects.py

```diff
@@ -0,0 +1,372 @@
+import xmltodict
+from datetime import datetime
+from dataclasses import dataclass, asdict, astuple
+
+
+@dataclass
+class ServerObject:
+    """ The minimum attributes and functionality of a Tableau server object """
+    id: str
```

Comment:
In the comment about this class it would be helpful to understand why you do all the type conversions in `_post_init`

Reply:
Sure, I will explain that in here, thanks!